src/tools/rbd_mirror/image_sync/ObjectCopyRequest.cc

```diff
@@ -103,6 +103,8 @@ void ObjectCopyRequest<I>::send_read_object() {
 
   bool read_required = false;
   librados::ObjectReadOperation op;
+  std::map<uint64_t, uint64_t> extents;
+  bufferlist xbl;
```

Comment:
Local variable -- will be destructed before async operation completes. Update the tuple to provide storage for the extents and bufferlist.